content/en/docs/concepts/storage/volume-snapshots.md

```diff
@@ -141,13 +142,44 @@ spec:
   driver: hostpath.csi.k8s.io
   source:
     snapshotHandle: 7bdd0de3-aaeb-11e8-9aae-0242ac110002
+  sourceVolumeMode: Filesystem
   volumeSnapshotRef:
     name: new-snapshot-test
     namespace: default
 ```
 
 `snapshotHandle` is the unique identifier of the volume snapshot created on the storage backend. This field is required for the pre-provisioned snapshots. It specifies the CSI snapshot id on the storage system that this `VolumeSnapshotContent` represents.
 
+`sourceVolumeMode` is the mode of the volume whose snapshot is taken. The value of the `sourceVolumeMode` field can be either `Filesystem` or `Block`. If the source volume mode is not specified, Kubernetes treats the snapshot as if the source volume's mode is unknown. Support for this field can only be enabled in VolumeSnapshot client v6.0.0 and higher. Visit the [VolumeSnapshot Release page](https://github.com/kubernetes-csi/external-snapshotter/releases) for more information.
+
+## Converting the volume mode of a snapshot {#convert-volume-mode}
+
+This feature is only present for `VolumeSnapshotContents` created with client version `v6.0.0` onwards.
```

Comment:
> This feature is only present for `VolumeSnapshotContents` created with client version `v6.0.0` onwards.

I'm guessing that a client can examine the `volumesnapshotcontents` API to see if it includes a `sourceVolumeMode` field; is that the only way for a client to tell whether the feature is supported?

Reply:
Only speaking for myself here, but I think this would be good to elaborate @RaunakShah .

Reply:
How about this: `This feature is in go module github.com/kubernetes-csi/external-snapshotter/client/v6 and later.`
I think a user can check if the new field `sourceVolumeMode` exists in a VolumeSnapshotContent spec or not by checking the manifest.

Reply:
In these docs, we almost never talk about Go modules. Do end users need to look at that level of detail to make use of snapshots?

Imagine someone is doing development with a bit of operations, and their thoughts run like this:
- I have access to a Kubernetes cluster cluster
- There is a team who look after the platform, including that cluster
- I want to know whether the cluster lets me use `sourceVolumeMode`
- I know I can write a test to see if `sourceVolumeMode` works in my cluster and then use the result of that test
- I know I can ask the platform team to do the same test and let me know the outcome
- if `sourceVolumeMode` is available, I want to use it. But only in the development environment.

Reply:
Can that developer write code that uses the Kubernetes API to work out whether `sourceVolumeMode` is available (eg by querying the cluster's OpenAPI data)?

If they can, let's give them a clear hint about that. Much better than to tell them to read the source code for a snapshotting component.
If the easiest way to work out what's available is to look at the Pod template for an add-on (`external-snapshotter`?) then we should tell them what to look for in that Pod template.

Reply:
They can check if `sourceVolumeMode` is available in the manifest using kubectl: `kubectl get crd volumesnapshotcontent -o yaml`.
@RaunakShah Can you update the PR with this?